### PR TITLE
[NFC] Deprecate LinalgExt TransformOps.

### PR DESCRIPTION
**Pull request by hanhanW**
These patterns and operations are not used in IREE, and they cause maintenance burden during integrate. E.g.,  https://github.com/openxla/iree/pull/12264, https://github.com/openxla/iree/pull/13338, ...

### CODE REVIEW

File: llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/TransformOps/LinalgExtTransformOps.td [hunk 1]
```diff
@@ -1,104 +0,0 @@
...
-
-  let extraClassDeclaration = [{
-    ::mlir::DiagnosedSilenceableFailure applyToOne(
-        ::mlir::scf::ForallOp target,
-        ::mlir::transform::ApplyToEachResultList &results,
-        ::mlir::transform::TransformState &state);
-  }];
-}
-
-def RewriteForallToScfForOp :
```
Comment (nicolasvasilache) [inline llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/TransformOps/LinalgExtTransformOps.td:68]: I am unclear this and the one above have been upstreamed yet ?

Comment (hanhanW) [reply llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/TransformOps/LinalgExtTransformOps.td:68]: Maybe I was wrong.. I've seen that there are many transform ops upstream, and some of them are about fusion. E.g.,

```
transform.sequence failures(propagate) {
^bb1(%arg1: !pdl.operation):
  %0 = transform.structured.match ops{["linalg.elemwise_binary"]} in %arg1 : (!pdl.operation) -> !pdl.operation
  %1, %loops:2 = transform.structured.fuse %0 {tile_sizes = [32, 32], tile_interchange = [0, 1]}
}
```

I thought they are not used by any transform works in IREE, so I assumed that all of them are upstreamed. My intention is having less unused things for better maintenance. Do we want to keep these operations? We can definitely keep them if they are still used in our work.

**Review by benvanik — APPROVED** [review]

**Comment by MaheshRavishankar** [comment]
Waiting for resolution here.

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
Blocking till Nicolas has a chance.

**Comment by hanhanW** [comment]
There are many conflicts. I can resolve them if we decide to delete the code.

**Pull request closed without merging**